### ISSUE TITLE
Auto quality for video streams [feature request]

### ISSUE THREAD

**Issue by stivens13** [lhc70000/iina#1432]
IINA's ability to play videos from internet is marvelous. However, it has a downside to native web players in sense of reliability of play. Specifically, those players have dynamic optimization regrading user's connection and continuous optimisation for smooth playback.

I propose support for 
a) choosing quality of video being played
b) auto mode when quality is dynamically changed referencing user's connection so no lags and bufferisation happen.

Is it possible in nearest future?

**Comment by Miigon** [comment]
Those web players (like Youtube) dynamically switch among **different sources**(different quality option) regarding user's connection. 'Different sources' means they have different URLs. The way IINA plays internet resources is to get the URL of the stream and play it. Which means IINA only knows **one** URL while playing internet resources. And IINA doesn't know how the original web players switch between different sources. Probably this feature won't be implemented.

**Comment by atomicsaiko** [comment]
Provided the streams aren't encrypted and offer multiple renditions this could be a nice feature for IINA. It would require IINA to support the parsing of common streaming formats like HLS and DASH and monitor networking conditions against the local playback buffer. The provided playlist show provide the info how to make the approcate HTTP GET requests for the different renditions of the same content.

**Comment by elmehalawi** [comment]
I agree this would be a great feature. A simpler "workaround" though would be to add a selection menu to the OpenInIINA browser extension to allow selecting a quality.

**Comment by LuckyDucko** [comment]
I agree with @elmehalawi, it would be great to have the ability to select what quality version you would like in some form of drop down in the browser extension.
